app/models/concerns/waste_carriers_engine/can_change_registration_status.rb

```diff
@@ -100,9 +99,24 @@ def should_not_be_expired?
       expiry_day = expiry_time_adjusted_for_daylight_savings.to_date
       # We store dates and times in UTC, but want to use the current date in the UK, not necessarily UTC
       current_day = Time.now.in_time_zone("London").to_date
+
+      return true if in_expiry_grace_window?(expiry_day, current_day)
+
       current_day < expiry_day
     end
 
+    # Its important to note that a registration is expired on its expires_on date.
+    # For example if the expires_on date is Oct 1, then the registration was
+    # ACTIVE Sept 30, and EXPIRED Oct 1. If the grace window is 3 days, just
+    # adding 3 days to that date would give the impression the grace window lasts
+    # till Oct 4 (i.e. 1 + 3) when in fact we need to include the 1st as one of
+    # our grace window days.
+    def in_expiry_grace_window?(expiry_date, current_day)
+      return false if current_day > ((expiry_date + Rails.configuration.grace_window) - 1.day)
```

Comment:
Think this line would be clearer if we calculated `((expiry_date + Rails.configuration.grace_window) - 1.day)` on a separate line and then compared the result to `current_day`, similar to the frontend: https://github.com/DEFRA/waste-carriers-frontend/pull/172/files#diff-85bbbdcab8cf9243156faf3f9a3873b8R44